lib/spack/llnl/util/cpu/microarchitecture.py

```diff
@@ -219,15 +231,10 @@ def optimization_flags(self, compiler, version):
         def satisfies_constraint(entry, version):
             min_version, max_version = entry['versions'].split(':')
 
-            # Check version suffixes
-            min_version, min_suffix = version_components(min_version)
-            max_version, max_suffix = version_components(max_version)
-            version, suffix = version_components(version)
-
-            # If the suffixes are not all equal there's no match
-            if ((suffix != min_suffix and min_version) or
-                (suffix != max_suffix and max_version)):
-                return False
+            # Extract numeric part of the version
+            min_version, _ = version_components(min_version)
+            max_version, _ = version_components(max_version)
+            version, _ = version_components(version)
```

Comment:
@tgamblin Just want to point out that this PR introduces the same change as https://github.com/archspec/archspec/pull/19 which is still under discussion.